src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -159,35 +155,101 @@ public void visitJavadocToken(DetailNode ast) {
         for (DetailNode child : ast.getChildren()) {
-            if (child.getType() == JavadocTokenTypes.TEXT) {
-                if (!CommonUtils.isBlank(child.getText())) {
-                    extraTextFound = true;
-                }
+            if (allowedTypes.contains(child.getType())) {
+                result.append(child.getText());
+            }
+            else if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG
+                    && getContentOfChild(child).equals(INHERIT_DOC)) {
+                result.append(INHERIT_DOC);
             }
-            else if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
-                if (child.getChildren()[1].getType() == JavadocTokenTypes.INHERIT_DOC_LITERAL) {
-                    containsInheritDoc = true;
-                }
-                else {
-                    extraTextFound = true;
-                }
+            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT
+                    && CommonUtils.isBlank(result.toString().trim())) {
+                getStringInsideTag(result, child.getChildren()[0].getChildren()[0]);
             }
-            else if (!SKIP_TOKENS.contains(child.getType())) {
-                extraTextFound = true;
+            else if (child.getType() == JavadocTokenTypes.JAVADOC_TAG) {
+                flag = false;
             }
-            if (extraTextFound) {
+            else if (isPeriodAtEnd(result, child)) {
+                periodFound = true;
+            }
+            if (!flag) {
                 break;
             }
         }
-        return containsInheritDoc && !extraTextFound;
+        printWarningForViolation(ast, periodFound, result);
+    }
+
+    /**
+     * Returns content when token type is javadoc inline tag.
+     * @param child javadoc inline tag ast.
+     * @return content of child nodes as string.
+     */
+    private static String getContentOfChild(DetailNode child) {
+        final StringBuilder contents = new StringBuilder();
+        for (DetailNode node : child.getChildren()) {
+            contents.append(node.getText().trim());
+        }
+        return contents.toString();
+    }
+
+    /**
+     * Concatenates string within text of html tags.
+     * @param result javadoc string
+     * @param detailNode javadoc tag node
+     */
+    private void getStringInsideTag(StringBuilder result, DetailNode detailNode) {
+        DetailNode tempNode = detailNode;
+        while (tempNode != null) {
+            if (tempNode.getType() == JavadocTokenTypes.TEXT) {
+                result.append(tempNode.getText());
+            }
+            tempNode = JavadocUtils.getNextSibling(tempNode);
+        }
+    }
+
+    /**
+     * Checks if period is set at end of first sentence.
+     * @param result javadoc string
+     * @param child child node
+     * @return true if period is at end of first sentence.
+     */
+    private boolean isPeriodAtEnd(StringBuilder result, DetailNode child) {
```

Comment:
argument should be String, `result.toString().trim()` should be done outside of this method.